src/CommunityToolkit.Maui/Behaviors/AnimationBehavior.shared.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Threading.Tasks;
+using System.Windows.Input;
+using CommunityToolkit.Maui.Animations;
+using Microsoft.Maui.Controls;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+/// <summary>
+/// The <see cref="AnimationBehavior"/> is a behavior that shows an animation on any <see cref="View"/> when the <see cref="AnimateCommand"/> is called.
+/// </summary>
+public class AnimationBehavior : EventToCommandBehavior
+{
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="AnimationType"/> property.
+	/// </summary>
+	public static readonly BindableProperty AnimationTypeProperty =
+		BindableProperty.Create(nameof(AnimationType), typeof(AnimationBase), typeof(AnimationBehavior));
+
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="AnimateCommand"/> property.
+	/// </summary>
+	internal static readonly BindablePropertyKey AnimateCommandPropertyKey =
+ 			BindableProperty.CreateReadOnly(
+ 				nameof(AnimateCommand),
+ 				typeof(ICommand),
+ 				typeof(AnimationBehavior),
+ 				default,
+ 				BindingMode.OneWayToSource,
+ 				defaultValueCreator: CreateAnimateCommand);
+	
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="AnimateCommand"/> property.
+	/// </summary>
+	public static readonly BindableProperty AnimateCommandProperty = AnimateCommandPropertyKey.BindableProperty;
+
+	/// <summary>
+	/// The type of animation to perform
+	/// </summary>
+	public AnimationBase? AnimationType
+	{
+		get => (AnimationBase?)GetValue(AnimationTypeProperty);
+		set => SetValue(AnimationTypeProperty, value);
+	}
+
+	/// <summary>
+	/// Command on which to perform the animation.
+	/// </summary>
+	public ICommand AnimateCommand => (ICommand)GetValue(AnimateCommandProperty);
+
+	bool isAnimating;
+	TapGestureRecognizer? tapGestureRecognizer;
+
+	/// <inheritdoc/>
+	protected override void OnAttachedTo(VisualElement bindable)
+	{
+		base.OnAttachedTo(bindable);
+
+		if (!string.IsNullOrWhiteSpace(EventName) || !(bindable is View view))
+			return;
+
+		tapGestureRecognizer = new TapGestureRecognizer();
+		tapGestureRecognizer.Tapped += OnTriggerHandled;
+		view.GestureRecognizers.Clear();
+		view.GestureRecognizers.Add(tapGestureRecognizer);
+	}
```

Comment:
Hi @brminnick, I've run into a bit of confusion when a TapGestureRecognizer is added to an Entry without an EventName specified. Maybe you have an idea of what's going on 😉

I would expect the entry to be editable and not trigger an animation by itself, but it triggers when I try to edit the text, making the entry uneditable. (see video)

Since this is basically a 1:1 from XCT, I compared it with the sample there, and the same scenario with an Entry without an EventName does work as expected and is editable. 

I'm confused because a TapGestureRecognizer is added in both cases, but it doesn't seem to do anything with an entry when running on xamarin forms but it does with MAUI 🤔

The only solution I can think of is adding an extra check to see what type of view it is and not adding a GR if it's something editable.  

https://user-images.githubusercontent.com/15085413/141359141-f545e23c-0cb4-448c-bd18-96e8c388d8d3.mp4

Reply:
Hey @stefanbogaard86! To be honest, I'm not quite sure.

The good news is that we don't have to have an exact 1:1 copy/paste from XCT. In fact, I highly recommend making changes to things that don't make sense. 

Once you've moved the PR out of "draft", we can give it a review 👍